tests/test_e2e_80_pathfinder.py

```diff
@@ -0,0 +1,160 @@
+                "link_name": "s2-eth4-s6-eth4",
+                "ownership": "blue",
+                "bandwidth": "100",
+                "delay" : 10,
+                "priority": 5,                
+            },
+            
+        }
+
+        for link_id, metadata in links_metadata.items():
+            api_url = f"{KYTOS_API}/topology/v3/links/{link_id}/metadata"
+            response = requests.post(
+                api_url,
+                data=json.dumps(metadata),
+                headers={"Content-type": "application/json"},
+            )
+            assert response.status_code == 201, response.text
+        return links_metadata        
+
+    @pytest.mark.parametrize("undesired_link, expected", 
+                            [([], 1),(["74bbc9527a0e309a86c95744042bcf9e3beb52955c942cac5fc735b1cf986f7f"], 8)])
+    def test_10_undesired_link_and_max_path(self, undesired_link, expected):
+        """Tests fastest path from switch 1 to 6 then blocks it"""
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"undesired_links": undesired_link,
+"spf_attribute": "hop",
+"spf_max_paths": 1,
+"parameter": "hop"
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['cost'] == expected, f'Shortest path expected {expected}: {data}'
+
+    @pytest.mark.parametrize("attribute, expected_cost", 
+                            [("hop", 1), ("delay", 20), ("priority", 15)])
+    def test_20_spf_attribute_and_max_path_cost(self, attribute, expected_cost):
+        """Tests hop, delay, and priority with path cost"""
+        links_metadata = self.add_topology_metadata()
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"spf_attribute": attribute,
+"spf_max_path_cost": expected_cost,
+"parameter": attribute
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['cost'] == expected_cost, f'Path cost larger than {expected_cost}: {data}'
+
+    def test_30_mandatory_metrics(self):
+        """The returned path should be of ownership blue"""
+        links_metadata = self.add_topology_metadata()
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"spf_attribute": "hop",
+"mandatory_metrics": {
+"ownership": "blue"
+}
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['metrics']['ownership'] == "blue", f'Path ownership not blue: {data}'
```

Comment:
Same as prior comments, to also assert the number of expected paths, and assert the hops/nodes of the minimum expected path too.